src/components/home/home.component.spec.ts

```diff
@@ -1,21 +1,32 @@
 import { ComponentFixture, TestBed } from '@angular/core/testing';
 import { HomeComponent } from './home.component';
+import { ProductService } from '../../services/products/product.service';
+
 
 describe('HomeComponent', () => {
   let component: HomeComponent;
   let fixture: ComponentFixture<HomeComponent>;
+  let productService: ProductService;
 
   beforeEach(async () => {
     await TestBed.configureTestingModule({
-      imports: [HomeComponent]
+      declarations: [HomeComponent],
+      providers: [
+        { provide: ProductService, useValue: jasmine.createSpyObj('ProductService', ['getList']) },
+      ],
     }).compileComponents();
+  });
 
+  beforeEach(() => {
     fixture = TestBed.createComponent(HomeComponent);
     component = fixture.componentInstance;
+    productService = TestBed.inject(ProductService);
+    spyOn(component, 'getProducts').and.callThrough();
```

Comment:
and yes, to verify if a specific endpoint is calling through the spyOn is the correct method...